thinc/legacy/loss.py

```diff
@@ -0,0 +1,230 @@
+from ..util import to_categorical, get_array_module
+from typing import Optional, Sequence, Dict, Union, Tuple
+from typing import TypeVar, Generic, Any, cast, List
+from ..types import Floats2d, Ints1d
+
+
+LossT = TypeVar("LossT")
+GradT = TypeVar("GradT")
+GuessT = TypeVar("GuessT")
+TruthT = TypeVar("TruthT")
+IntsOrFloats = Union[Ints1d, Floats2d]
+IntsOrFloatsOrStrs = Union[Ints1d, Floats2d, Sequence[int], Sequence[str]]
+
+
+class Loss(Generic[GuessT, TruthT, GradT, LossT]):  # pragma: no cover
```

Comment:
I don't think we have to repeat this class definition, do we? I would put in `legacy` only the parts that have changed in newer versions

Reply:
Oh, I've put stuff here only because `loss.py` imports from `legacy/loss.py` and then `legacy/loss.py` would import from `loss.py` and that would create a circular import.

Reply:
Right, but I think we can avoid the import from `legacy/loss.py` in `loss.py` and basically define the v1 etc directly in  `legacy/loss.py`?